aws/resource_aws_rds_cluster_endpoint.go

```diff
@@ -0,0 +1,267 @@
+		return fmt.Errorf("error setting static_members: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsRDSClusterEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	input := &rds.ModifyDBClusterEndpointInput{
+		DBClusterEndpointIdentifier: aws.String(d.Id()),
+	}
+
+	if v, ok := d.GetOk("custom_endpoint_type"); ok {
+		input.EndpointType = aws.String(v.(string))
+	}
+
+	if attr := d.Get("excluded_members").(*schema.Set); attr.Len() > 0 {
+		input.ExcludedMembers = expandStringList(attr.List())
+	} else {
+		input.ExcludedMembers = make([]*string, 0)
+	}
+
+	if attr := d.Get("static_members").(*schema.Set); attr.Len() > 0 {
+		input.StaticMembers = expandStringList(attr.List())
+	} else {
+		input.StaticMembers = make([]*string, 0)
+	}
+
+	_, err := conn.ModifyDBClusterEndpoint(input)
+	if err != nil {
+		return fmt.Errorf("Error modifying RDS Cluster Endpoint: %s", err)
+	}
+
+	return resourceAwsRDSClusterEndpointRead(d, meta)
+}
+
+func resourceAwsRDSClusterEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	input := &rds.DeleteDBClusterEndpointInput{
+		DBClusterEndpointIdentifier: aws.String(d.Id()),
+	}
+	_, err := conn.DeleteDBClusterEndpoint(input)
+	if err != nil {
+		return fmt.Errorf("Error deleting RDS Cluster Endpoint: %s", err)
+	}
+
+	if err := resourceAwsRDSClusterEndpointWaitForDestroy(d.Timeout(schema.TimeoutDelete), d.Id(), conn); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsRDSClusterEndpointWaitForDestroy(timeout time.Duration, id string, conn *rds.RDS) error {
+	log.Printf("Waiting for RDS Cluster Endpoint %s to be deleted...", id)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"available", "deleting"},
+		Target:     []string{"destroyed"},
+		Refresh:    DBClusterEndpointStateRefreshFunc(conn, id),
+		Timeout:    timeout,
+		Delay:      AWSRDSClusterEndpointRetryDelay,
+		MinTimeout: AWSRDSClusterEndpointRetryMinTimeout,
+	}
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for RDS Cluster Endpoint (%s) to be deleted: %v", id, err)
+	}
+	return nil
+}
+
+func resourceAwsRDSClusterEndpointWaitForAvailable(timeout time.Duration, id string, conn *rds.RDS) error {
+	log.Printf("Waiting for RDS Cluster Endpoint %s to become available...", id)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating"},
+		Target:     []string{"available"},
+		Refresh:    DBClusterEndpointStateRefreshFunc(conn, id),
+		Timeout:    timeout,
+		Delay:      AWSRDSClusterEndpointRetryDelay,
+		MinTimeout: AWSRDSClusterEndpointRetryMinTimeout,
+	}
+
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for RDS Cluster Endpoint (%s) to be ready: %v", id, err)
+	}
+	return nil
+}
+
+func DBClusterEndpointStateRefreshFunc(conn *rds.RDS, id string) resource.StateRefreshFunc {
```

Comment:
Nit: In Go, capitalized function names mean they are exported and available for usage outside the current Go package (e.g. `public` in Java). While we currently don't make any compatibility promises about using this codebase as a library, we should avoid exporting functions where possible. I believe this will be caught by `golint` in the future (if we enable it in this codebase) since as its written this exported function does not have an associated comment.